### DISCUSSION TITLE
RandomPointsBuilder.CreateRandomCoord

### DISCUSSION THREAD

**Discussion by upswing1** [Mapsui/Mapsui#2067] [Q&A]
I need help trying to replicate code found In Demo 2 MapInfo
I have the following error:
```console
Error	CS0117	'RandomPointsBuilder' does not contain a definition for 'CreateProviderWithRandomPoints'	
```
in method

```csharp
private static ILayer CreateInfoLayer(MRect? envelope)
{
    var random = new Random(7);

    return new Layer(InfoLayerName)
    {
        DataSource = RandomPointsBuilder.CreateProviderWithRandomPoints(envelope, 25, random),
        Style = CreateSymbolStyle(),
        IsMapInfoLayer = true
    };
}
```

**Comment by pauldendulk** [comment]
RandomPointsBuilder.CreateProviderWithRandomPoints is one of the helper methods we use in our samples. You could create something like that yourself or you could copy that class from the Mapsui.Samples.Common project.

**Reply by pauldendulk** [reply]
Then again. It would be nice if the code samples would just work with a simple copy and paste, without having to retrieve used classes. Perhaps we should copy more code to the sample classes. Perhaps we could add some helpers in the Builders we intend to add.

**Reply by upswing1** [reply]
I agree, having a simple copy and paste in the code samples will make the code samples easier to learn.
